packages/ra-core/src/types.ts

```diff
@@ -72,54 +72,210 @@ export type LegacyAuthProvider = (
  * dataProvider types
  */
 
-export type DataProvider<ResourceType extends string = string> = {
-    getList: <RecordType extends RaRecord = any>(
+// https://stackoverflow.com/questions/51465182/how-to-remove-index-signature-using-mapped-types/68261113#68261113
+type RemoveIndex<T> = {
+    [P in keyof T as string extends P
```

Comment:
Fixed by updating Prettier: #7365

Reply:
To make this mergeable in standalone, can you add an ignore comment?